release-notes/swan-lake-GA-release-note.md

```diff
@@ -27,10 +27,237 @@ If you have not installed Ballerina, then download the [installers](/downloads/#
+
+```ballerina
+function fn(Employee|Student v) {
+   if v is Employee {
+       // `v` is narrowed to `Employee` here.
+       string dept = v.department;
+   } else {
+       // `v` was previously narrowed to `Student` here.
+       // Will now result in a compilation error.
+       int grade = v.grade;
+   }
+}
+```
+
+Even though jBallerina currently allows only values that belong to either `Employee` or `Student` to be passed as arguments for the parameter of type `Employee|Student`, the Ballerina specification defines subtyping to be semantic. This along with mutability would mean that a value that does not belong to `Employee` nor `Student` but belongs to `Employee|Student` can be passed as an argument here, which results in the possibility of the value not belonging to `Student` in the else block.
+
+For example, one would be able to call `fn` with the following.
+
+```ballerina
+public function main() {
+   record {|
+       int|string id;
+       string name;
+       anydata grade;
+       anydata department;
+   |} rec = {
+       id: "A1234",
+       name: "Amy",
+       department: ["physics", 1],
+       grade: 12
+   };
+   fn(rec);
+}
+```
+
+Although it is not possible to call this function in this manner at the moment since jBallerina does not support semantic subtyping, the changes to narrowing have been introduced in this release to minimize future incompatibility issues.
+
+- A spec deviation which allowed non-required fields of records/maps and error detail records/maps to be bound using mapping and error binding patterns in variable declarations and destructuring assignment statements has been fixed. Attempting to bind a non-required field will now result in a compilation error.
+
+```ballerina
+type Employee record {
+   string name;
+   int age?;
+};
+ 
+function bindEmployee(Employee emp) {
+   string empName;
+   int empAge;
+ 
+   // The 'age' field is optional in the `Employee` record type-descriptor.
+   // Therefore, attempting to bind it will now result in compilation errors.  
+   {name: empName, age: empAge} = emp; // error for `age: empAge`
+   var {name: empNameOne, age: empAgeOne} = emp; // error for `age: empAgeOne`
+ 
+   // Similarly, since there is no required `department` field in the `Employee`
+   // record, the following will also result in an error.
+   var {name, department} = emp; // error for `department`
+}
+ 
+type Error error<record {| int code?; string identifier; boolean...; |}>;
+ 
+function bindError(Error err) {
+   // The 'code' field is optional in the detail record of `Error`.
+   // Therefore, attempting to bind it will now result in a compilation error.
+   Error error(message1, code = code1, identifier = identifier1) = err; // error for 'code = code1'
+ 
+   // Similarly, attempting to bind an undefined (non-required) `fatal` field will
+   // also result in a compilation error.
+   var error(message, fatal = fatal) = err; // error for 'fatal = fatal'
+}
+```
```

Comment:
````suggestion
  ```ballerina
  type Employee record {
     string name;
     int age?;
  };
   
  function bindEmployee(Employee emp) {
     string empName;
     int empAge;
   
     // The 'age' field is optional in the `Employee` record type-descriptor.
     // Therefore, attempting to bind it will now result in compilation errors.  
     {name: empName, age: empAge} = emp; // error for `age: empAge`
     var {name: empNameOne, age: empAgeOne} = emp; // error for `age: empAgeOne`
   
     // Similarly, since there is no required `department` field in the `Employee`
     // record, the following will also result in an error.
     var {name, department} = emp; // error for `department`
  }
   
  type Error error<record {| int code?; string identifier; boolean...; |}>;
   
  function bindError(Error err) {
     // The 'code' field is optional in the detail record of `Error`.
     // Therefore, attempting to bind it will now result in a compilation error.
     Error error(message1, code = code1, identifier = identifier1) = err; // error for 'code = code1'
   
     // Similarly, attempting to bind an undefined (non-required) `fatal` field will
     // also result in a compilation error.
     var error(message, fatal = fatal) = err; // error for 'fatal = fatal'
  }
  ```

````